cns/restserver/api.go

```diff
@@ -1415,3 +1415,44 @@ func (service *HTTPRestService) deleteHostNCApipaEndpoint(w http.ResponseWriter,
 	err = service.Listener.Encode(w, &response)
 	logger.Response(service.Name, response, response.Response.ReturnCode, ReturnCodeToString(response.Response.ReturnCode), err)
 }
+
+// This function is used to query NMagents's supproted APIs list
+func (service *HTTPRestService) nmAgentSupportedApisHandler(w http.ResponseWriter, r *http.Request) {
+	logger.Request(service.Name, "nmAgentSupportedApisHandler", nil)
+	var (
+		err           error
+		req           cns.NmAgentSupportedApisRequest
+		returnCode    int
+		returnMessage string
+		supportedApis []string
+	)
+
+	err = service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		return
+	}
+
+	switch r.Method {
+	case http.MethodPost:
```

Comment:
Could this be a GET? After all, the client is getting the supported APIs from CNS, and CNS knows how to communicate to nmagent. Do we need to pass a url in the body?

Reply:
We had to change from IMDS to Wireserver, what if we want to change this again, albeit long odds.

Reply:
DNC -> CNS POST message with payload on what CNS needs to send to wireserver. 
CNS-> Wireserver is GET